integration_tests/e2e/checkYourBooking.cy.ts

```diff
@@ -268,4 +268,127 @@ context('Check visit details page', () => {
+    cy.task('stubVisitSessions', {
+      offenderNo: prisonerId,
+      prisonId,
+      visitSessions,
+    })
+    cy.task('stubOffenderEvents', {
+      offenderNo: prisonerId,
+      fromDate: format(today, shortDateFormat),
+      toDate: format(addDays(today, 8), shortDateFormat),
+      scheduledEvents: [],
+    })
+    cy.task(
+      'stubCreateVisitApplication',
+      TestData.application({
+        startTimestamp: visitSessions[0].startTimestamp,
+        endTimestamp: visitSessions[0].endTimestamp,
+        visitors: [{ nomisPersonId: contacts[0].personId }],
+      }),
+    )
+    selectVisitorsPage.continueButton().click()
+    const selectVisitDateAndTime = Page.verifyOnPage(SelectVisitDateAndTime)
+    selectVisitDateAndTime.expandAllSections()
+    selectVisitDateAndTime.getSlotById(1).check()
+
+    // Additional support
+    selectVisitDateAndTime.continueButton().click()
+    const additionalSupportPage = Page.verifyOnPage(AdditionalSupportPage)
+    additionalSupportPage.additionalSupportNotRequired().check()
+
+    // Main contact
+    additionalSupportPage.continueButton().click()
+    const mainContactPage = Page.verifyOnPage(MainContactPage)
+    mainContactPage.getFirstContact().check()
+    mainContactPage.phoneNumberTrueRadio().click()
+    mainContactPage.enterPhoneNumber('01234 567890')
+    cy.task(
+      'stubChangeVisitApplication',
+      TestData.application({
+        startTimestamp: visitSessions[0].startTimestamp,
+        endTimestamp: visitSessions[0].endTimestamp,
+        visitContact: { name: 'Jeanette Smith', telephone: '01234 567890' },
+        visitors: [{ nomisPersonId: contacts[0].personId, visitContact: true }],
+        visitorSupport: { description: '' },
+        sessionTemplateReference: visitSessions[0].sessionTemplateReference,
+      }),
+    )
+
+    // Request method
+    mainContactPage.continueButton().click()
+    const requestMethodPage = Page.verifyOnPage(RequestMethodPage)
+    requestMethodPage.getRequestLabelByValue('PHONE').contains('Phone call')
+    requestMethodPage.getRequestMethodByValue('PHONE').check()
+    requestMethodPage.continueButton().click()
+
+    // Check visit details
+    const checkYourBookingPage = Page.verifyOnPage(CheckYourBookingPage)
+    checkYourBookingPage.prisonerName().contains(prisonerDisplayName)
+    checkYourBookingPage.visitDate().contains(format(new Date(visitSessions[0].startTimestamp), longDateFormat))
+
+    cy.task('stubBookVisitValidationFailed', {
+      visit: TestData.visit({
+        startTimestamp: visitSessions[1].startTimestamp,
+        endTimestamp: visitSessions[1].endTimestamp,
+        visitContact: { name: 'Jeanette Smith', telephone: '09876 543 321' },
+        visitors: [
+          { nomisPersonId: contacts[0].personId, visitContact: true },
+          { nomisPersonId: contacts[1].personId, visitContact: false },
+        ],
+        visitorSupport: { description: 'Wheelchair ramp' },
+      }),
```

Comment:
Could just pass an `applicationReference` here - see comment in `orchestration.ts`.